NumLib/Extrapolation/LocalLinearLeastSquaresExtrapolator.cpp

```diff
@@ -271,6 +333,102 @@ void LocalLinearLeastSquaresExtrapolator::extrapolateElement(
     }
 }
 
+void LocalLinearLeastSquaresExtrapolator::extrapolateElement(
+    std::size_t const element_index,
+    const unsigned num_components,
+    ExtrapolatableElementCollection const& extrapolatables,
+    GlobalVector const& int_pt_x,
+    std::vector<std::vector<GlobalIndexType>>& chemical_system_index_map,
+    GlobalVector& counts)
+{
+    assert(num_components == 1);
+    _integration_point_values_cache.clear();
+    auto const& indices = chemical_system_index_map[element_index];
+    _integration_point_values_cache = int_pt_x.get(indices);
```

Comment:
Better to use `getIntegrationPointValues` interface of extrapolatables to get the values, instead of passing a single vector of those values.
In the process you have a reference to the `GlobalVector int_pt_x` and you get the element number through the `getIntegrationPointValues` interface. From these two and the "chemical_system_index_map" (which must be part of the process, otherwise you couldn't pass it here) you get the `_integration_point_values_cache` filled and all of the duplication in the NumLib is not necessary.

Reply:
I was looking for a workaround to get rid of duplications, but no good idea comes up with me. Thanks for the thoughtful suggestion! I will redo the extrapolation in the way you suggested.